glusterd2/store/store.go

```diff
@@ -125,3 +128,33 @@ func (s *GDStore) UpdateEndpoints() error {
 	s.conf.Endpoints = s.Client.Endpoints()
 	return s.conf.Save()
 }
+
+// Returns a new GDStore from the given etcd Client, with namespaced KV, Lease, Watcher, Session etc.
+func newNamespacedStore(oc *clientv3.Client, conf *Config) (*GDStore, error) {
+	namespaceKey := fmt.Sprintf("gluster-%s/", gdctx.MyClusterID.String())
+
+	// Create namespaced interfaces
+	kv := namespace.NewKV(oc.KV, namespaceKey)
+	lease := namespace.NewLease(oc.Lease, namespaceKey)
+	watcher := namespace.NewWatcher(oc.Watcher, namespaceKey)
+
+	// Create a new session (lease kept alive for the lifetime of a client)
```

Comment:
This comment belongs just before the call to `concurrency.NewSession`

Reply:
Nope. It's in the right place. The two blocks below together create the proper new session.